src/vocab/subword.rs

```diff
@@ -190,18 +176,15 @@ where
             .map(|ngram| ngram.to_string())
             {
                 let cnt = ngram_counts.entry(ngram).or_default();
-                *cnt += count;
+                *cnt += word.count();
             }
-            words.push(word);
         }
-
         let mut ngrams = ngram_counts
             .iter()
             .filter(|(_, count)| **count >= config.indexer.min_ngram_count as usize)
             .map(|(ngram, _)| ngram.clone())
             .collect::<Vec<_>>();
 
```

Comment:
I think we should also implement target vocab size for the ngrams, although the token-vocabs using `CountedType`s and the explicit indexer taking `String`s complicates things a bit. We're not actually using counts after calculating the discard probabilities, so maybe we can get rid of `CountedType` at some point.

I think this goes beyond the scope of this PR and we should discuss this a bit more, just something to note.